GeneralsMD/Code/GameEngine/Source/Common/System/BuildAssistant.cpp

```diff
@@ -718,6 +705,19 @@ LegalBuildCode BuildAssistant::isLocationClearOfObjects( const Coord3D *worldPos
 		if (them->isKindOf(KINDOF_INERT))
 			continue;
 
+		if (builderObject && them->getShroudedStatus(builderObject->getControllingPlayer()->getPlayerIndex()) >= OBJECTSHROUD_FOGGED)
+			return LBC_SHROUD;
+
+		//Kris: Patch 1.01 - November 5, 2003
+		//Prevent busy units (black lotus hacking from being moved by trying to place a building -- exploit).
```

Comment:
Why did this block need moving? If it needs to be here for some reason, can add a comment to help understand.

Reply:
I thought it made more sense for the continues (the ignores) to come before the returns (the fails) - in the odd case something designated removable for construction, inert, a mine, etc. could be busy). And if we were to add a bib to this block as you suggested, it would make sense to process that logic after the shroud check to avoid the same issue the original fix was resolving.